src/audio/multiband_drc/multiband_drc.c

```diff
@@ -321,18 +321,42 @@ static int multiband_drc_cmd_set_value(struct processing_module *mod,
 }
 #endif
 
-static int multiband_drc_set_config(struct processing_module *mod, uint32_t config_id,
+static int multiband_drc_set_config(struct processing_module *mod, uint32_t param_id,
 				    enum module_cfg_fragment_position pos,
 				    uint32_t data_offset_size, const uint8_t *fragment,
 				    size_t fragment_size, uint8_t *response,
 				    size_t response_size)
 {
 	struct multiband_drc_comp_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
 
-	comp_dbg(mod->dev, "multiband_drc_set_config()");
+	comp_dbg(dev, "multiband_drc_set_config()");
 
-#if CONFIG_IPC_MAJOR_3
-	/* Other control types are possible only with IPC3 */
+#if CONFIG_IPC_MAJOR_4
+	struct sof_ipc4_control_msg_payload *ctl;
+
+	switch (param_id) {
+	case SOF_IPC4_SWITCH_CONTROL_PARAM_ID:
+		ctl = (struct sof_ipc4_control_msg_payload *)fragment;
+		comp_dbg(dev, "SOF_IPC4_SWITCH_CONTROL_PARAM_ID id = %d, num_elems = %d",
+			 ctl->id, ctl->num_elems);
+
+		if (ctl->id == 0 && ctl->num_elems == 1) {
+			cd->process_enabled = ctl->chanv[0].value;
+			comp_info(dev, "process_enabled = %d", cd->process_enabled);
+		} else {
+			comp_err(dev, "Illegal control id = %d, num_elems = %d",
+				 ctl->id, ctl->num_elems);
```

Comment:
but still returning 0 - ignoring and continuing?

Reply:
should not, need return a negative value.

Reply:
I'll test what happens if I return error, I didn't want to stop streaming so I left it out, but maybe it doesn't. It's good if only amixer errors.

Reply:
The control handler now errors as requested. The playback doesn't start at all since the control is sent to FW after init. I guess it's good to error incorrect topologies instead of just logging the errors, so proposing this now.